src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -279,32 +280,98 @@ public static TaskDescription parseTaskDescription(String name) throws ParseExce
     public static GradedTest parseGradedTest(String gradedTest) throws ParseException {
         requireNonNull(gradedTest);
         String trimmedGradedTest = gradedTest.trim();
+        if ("default".equalsIgnoreCase(gradedTest)) {
+            ReadingAssessment1 readingAssessment1 = new ReadingAssessment1(DEFAULT_VALUE);
+            ReadingAssessment2 readingAssessment2 = new ReadingAssessment2(DEFAULT_VALUE);
+            MidTerms midTerms = new MidTerms(DEFAULT_VALUE);
+            Finals finals = new Finals(DEFAULT_VALUE);
+            PracticalExam practicalExam = new PracticalExam(DEFAULT_VALUE);
 
-        try {
-            String[] components = trimmedGradedTest.split("\\|");
-
-            if (components.length != 5) {
-                throw new ParseException("Invalid GradedTest format. Expected 5 components.");
+            return new GradedTest(readingAssessment1, readingAssessment2, midTerms, finals, practicalExam);
+        } else {
+            try {
+                ReadingAssessment1 readingAssessment1 = new ReadingAssessment1(gradedTestParser(trimmedGradedTest)[0]);
+                ReadingAssessment2 readingAssessment2 = new ReadingAssessment2(gradedTestParser(trimmedGradedTest)[1]);
+                MidTerms midTerms = new MidTerms(gradedTestParser(trimmedGradedTest)[2]);
+                Finals finals = new Finals(gradedTestParser(trimmedGradedTest)[3]);
+                PracticalExam practicalExam = new PracticalExam(gradedTestParser(trimmedGradedTest)[4]);
+
+                return new GradedTest(readingAssessment1, readingAssessment2, midTerms, finals, practicalExam);
+            } catch (ParseException e) {
+                if (!GradedTest.isValidGradeTestName(trimmedGradedTest)) {
+                    throw new ParseException(GradedTest.MESSAGE_CONSTRAINTS);
+                }
+                return new GradedTest(trimmedGradedTest);
```

Comment:
test case

Reply:
Codecov is confusing :(
doesnt this test already cover that line?

```
    @Test
    public void parseGradedTest_invalidFormat_throwsParseException() {
        String input = "RA1:10 | RA2:20 | MidTerms:30 | Finals:40 | PE:50 | ExtraComponent:60";
        assertThrows(ParseException.class, () -> ParserUtil.parseGradedTest(input));
    }
```